components/server/src/user/token-service.ts

```diff
@@ -10,37 +10,43 @@ import { HostContextProvider } from "../auth/host-context-provider";
 import { UserDB } from "@gitpod/gitpod-db/lib";
 import { v4 as uuidv4 } from "uuid";
 import { TokenProvider } from "./token-provider";
+import { ApplicationError, ErrorCodes } from "@gitpod/gitpod-protocol/lib/messaging/error";
 
 @injectable()
 export class TokenService implements TokenProvider {
     static readonly GITPOD_AUTH_PROVIDER_ID = "Gitpod";
-    static readonly GITPOD_PORT_AUTH_TOKEN_EXPIRY_MILLIS = 30 * 60 * 1000;
 
     @inject(HostContextProvider) protected readonly hostContextProvider: HostContextProvider;
     @inject(UserDB) protected readonly userDB: UserDB;
 
     protected getTokenForHostCache = new Map<string, Promise<Token>>();
 
-    async getTokenForHost(user: User, host: string): Promise<Token> {
+    async getTokenForHost(user: User | string, host: string): Promise<Token> {
+        const userId = User.is(user) ? user.id : user;
         // (AT) when it comes to token renewal, the awaited http requests may
         // cause "parallel" calls to repeat the renewal, which will fail.
         // Caching for pending operations should solve this issue.
-        const key = `${host}-${user.id}`;
+        const key = `${host}-${userId}`;
```

Comment:
Should we check host? nowhere in server side check it

Reply:
@mustard-mh, sorry server related changes are part of https://github.com/gitpod-io/gitpod/pull/19098
Here it's only about the dashboard. But is still look into this comment in the other PR.
I had a stacked PR, and merged first, now the rest is mixed up. :-(